pkg/client/cache/expiration_cache_test.go

```diff
@@ -62,6 +62,59 @@ func TestTTLExpirationBasic(t *testing.T) {
 	close(deleteChan)
 }
 
+func TestReAddExpiredItem(t *testing.T) {
+	deleteChan := make(chan string, 1)
```

Comment:
This is tricky to test. As is the test will always pass, but it will fail with the old code and a sleep in the goroutine that triggers the race.